src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -866,6 +841,63 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         }
     }
 
+    /**
+     * Picks the cluster state with highest version from the queue. All cluster states with lower versions are ignored
+     */
+    static ClusterState selectMostRecentState(Queue<ProcessClusterState> processNewClusterStates) {
```

Comment:
ok that is good, I named that this way because most of the times the CS with the highest version is picked.